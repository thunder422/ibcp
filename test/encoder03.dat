######################################
#  Encoder Test #3: Dictionary Test  #
######################################
+Var% = 4
+var% = VAR% + 1
+REM comment
+REM Comment
+REM COMMENT
+Var$ = "Test"
+VAR$ = "TEST"
+var$ = "test"
+Var = 1.23e45 / 9.87E-4
+VAR = var * 1.23E45 / 9.87e-4
