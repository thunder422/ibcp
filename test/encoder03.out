
Program:
0: [0-4] 0:VarRefInt 1:|0:Var%| 2:ConstInt 3:|0:4| 4:=Assign%
1: [5-12] 0:VarRefInt 1:|0:Var%| 2:VarInt 3:|0:Var%| 4:ConstInt 5:|1:1| 6:+% 7:=Assign%
2: [13-14] 0:REM 1:|0: comment|
3: [15-16] 0:REM 1:|1: Comment|
4: [17-18] 0:REM 1:|2: COMMENT|
5: [19-23] 0:VarRefStr 1:|0:Var$| 2:ConstStr 3:|0:Test| 4:=Assign$
6: [24-28] 0:VarRefStr 1:|0:Var$| 2:ConstStr 3:|1:TEST| 4:=Assign$
7: [29-33] 0:VarRefStr 1:|0:Var$| 2:ConstStr 3:|2:test| 4:=Assign$
8: [34-41] 0:VarRef 1:|0:Var| 2:Const 3:|2:1.23e45| 4:Const 5:|3:9.87E-4| 6:/ 7:=Assign
9: [42-52] 0:VarRef 1:|0:Var| 2:Var 3:|0:Var| 4:Const 5:|2:1.23e45| 6:* 7:Const 8:|3:9.87E-4| 9:/ 10:=Assign

Remarks:
0: 1 | comment|
1: 1 | Comment|
2: 1 | COMMENT|
Free: none

Number Constants:
0: 1 |4|
1: 1 |1|
2: 2 |1.23e45|
3: 2 |9.87E-4|
Free: none

String Constants:
0: 1 |Test|
1: 1 |TEST|
2: 1 |test|
Free: none

Double Variables:
0: 3 |Var|
Free: none

Integer Variables:
0: 3 |Var|
Free: none

String Variables:
0: 3 |Var|
Free: none

Output:
0: Var% = 4
1: Var% = Var% + 1
2: REM comment
3: REM Comment
4: REM COMMENT
5: Var$ = "Test"
6: Var$ = "TEST"
7: Var$ = "test"
8: Var = 1.23e45 / 9.87E-4
9: Var = Var * 1.23e45 / 9.87E-4
